event-tracking/event-tracking-service/src/main/java/uk/gov/justice/services/subscription/ProcessedEventTrackingRepository.java

```diff
@@ -107,36 +109,47 @@ public Stream<ProcessedEvent> getAllProcessedEventsDescendingOrder(final String
     }
 
     @Transactional(REQUIRES_NEW)
-    public Stream<ProcessedEvent> getProcessedEventsLessThanEventNumberInDescendingOrder(
+    public List<ProcessedEvent> getProcessedEventsLessThanEventNumberInDescendingOrder(
             final Long fromEventNumber,
             final Long batchSize,
             final String source,
             final String componentName) {
 
+        final ArrayList<ProcessedEvent> processedEvents = new ArrayList<>();
+
         try {
-            final PreparedStatementWrapper preparedStatement = preparedStatementWrapperFactory.preparedStatementWrapperOf(
-                    viewStoreJdbcDataSourceProvider.getDataSource(), SELECT_LESS_THAN_EVENT_NUMBER_IN_DESCENDING_ORDER_SQL);
+            final Connection connection = viewStoreJdbcDataSourceProvider.getDataSource().getConnection();
+            final PreparedStatement preparedStatement = connection.prepareStatement(SELECT_LESS_THAN_EVENT_NUMBER_IN_DESCENDING_ORDER_SQL);
 
             preparedStatement.setString(1, source);
             preparedStatement.setString(2, componentName);
             preparedStatement.setLong(3, fromEventNumber);
             preparedStatement.setLong(4, batchSize);
 
-            return jdbcResultSetStreamer.streamOf(preparedStatement, resultSet -> {
-
-                try {
+            try (final ResultSet resultSet = preparedStatement.executeQuery()) {
+                while (resultSet.next()) {
```

Comment:
Been a long time since I looked at this stuff, so just wanting to check... This moves the pointer on if there is another row and it will be processed within the condition, we won't be skipping the last row as there isn't a 'next' one?

Reply:
So I make the query ordered by event number in descending order, This is because I don't know what the last event number is but I do know that the first event number is 1 (and its previous event number is zero). Each time the iteration happens I store the 'previous' event number as current event number, Therefore if current event number is zero, there are no more events and I can return false

Reply:
Also 'next()' on a result set moves the cursor forward one row. From the docs:
_Moves the cursor forward one row from its current position. A ResultSet cursor is initially positioned before the first row; the first call to the method next makes the first row the current row; the second call makes the second row the current row, and so on._

>